src/App.tsx

```diff
@@ -1,61 +1,131 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo, User } from './components/Types';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userId: number): User | null {
+  const userById = usersFromServer.find(user => user.id === userId);
+
+  return userById || null;
+}
+
+function getUserByName(name: string): User | null {
+  const userByName = usersFromServer.find(user => user.name === name);
+
+  return userByName || null;
+}
+
+export const receivedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [titleInput, setTitleInput] = useState('');
+  const [userSelect, setUserSelect] = useState('');
+  const [todos, setTodos] = useState(receivedTodos);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const handleNewTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!titleInput) {
+      setTitleError(true);
+
+      return;
+    }
+
+    if (!userSelect) {
+      setUserError(true);
+
+      return;
+    }
+
+    const receivedUser = getUserByName(userSelect);
+
+    const maxTodoId: number = [...todos]
+      .sort((todo1, todo2) => todo2.id - todo1.id)[0].id;
+
+    setTodos([
+      ...todos,
+      {
+        id: maxTodoId + 1,
+        title: titleInput,
+        completed: false,
+        userId: receivedUser?.id,
+        user: receivedUser,
+      },
+    ]);
+  };
```

Comment:
Let's create a separate variable for new todo to improve readability 

```suggestion
     const newTodo = {
        id: maxTodoId + 1,
        title: titleInput,
        completed: false,
        userId: receivedUser?.id,
        user: receivedUser,
    },
    setTodos([ ...todos, newTodo ]);
  };
```